File: design-hashset_705.py
class MyHashSet(object):
    # implemented s30 solution.
    def __init__(self):
        self.buckets = 1000
        self.bucketItems = 1000
        self.storage = [None] * self.buckets

    def getBucket(self, key):
        return key % self.buckets

    def getBucketItems(self, key):
        return key / self.bucketItems

    def add(self, key):
        """
        :type key: int
        :rtype: None
        """
        bucket = self.getBucket(key)
        bucketItem = self.getBucketItems(key)
        


        if self.storage[bucket] == None:
            if bucket == 0:
                self.storage[bucket] = [False] * (self.bucketItems + 1)
            else:
                self.storage[bucket] = [False] * self.bucketItems
        self.storage[bucket][bucketItem] = True

    def remove(self, key):
        """
        :type key: int
        :rtype: None
        """
        bucket = self.getBucket(key)
        bucketItem = self.getBucketItems(key)        
        if self.storage[bucket] == None:
            return
        self.storage[bucket][bucketItem] = False

    def contains(self, key):
        """
        :type key: int
        :rtype: bool
        """
        bucket = self.getBucket(key)
        bucketItem = self.getBucketItems(key)
        if self.storage[bucket] == None:
            return False
        
        return self.storage[bucket][bucketItem]     


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(1)
# obj.add(2)
# obj.remove(1)
# param_3 = obj.contains(1)

File: designhashset705.java
class MyHashSet {
    /*
    Time Complexity : O(1)
    Space Complexity : O(N) N = 1000000 
    Did this code successfully run on Leetcode : Yes
    Any problem you faced while coding this : No
    */
    boolean[][] set;

    public MyHashSet() {
        set = new boolean[1001][1001];
    }
    
    public void add(int key) {
        int position = getPosition(key);
        int bucket = getBucket(key);
        if(set[position]==null){
            set[position] = new boolean[1001];
        }
        set[position][bucket] = true;
        
    }
    
    public void remove(int key) {
        int position = getPosition(key);
        int bucket = getBucket(key);
        if(set[position]==null){
            return;
        }
        set[position][bucket]=false;
        
    }
    
    public boolean contains(int key) {
        int position = getPosition(key);
        int bucket = getBucket(key);
        if(set[position]==null){
            return false;
        }
        return set[position][bucket];
    }

    public int getPosition(int val){
        return val /1000;
    }
    public int getBucket(int val){
        return val % 1000;
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.add(key);
 * obj.remove(key);
 * boolean param_3 = obj.contains(key);
 */